python/tests/test_highlevel.py

```diff
@@ -4209,6 +4209,22 @@ def test_node_edges(self):
                 else:
                     assert edge == tskit.NULL
 
+    @pytest.mark.parametrize(
+        "t,expected_num_siblings",
+        [
+            (tskit.Tree.generate_random_binary(num_leaves=3, random_seed=1), 1),
```

Comment:
I don't think there's much point in having a random  tree here, just use ``generate_balanced`` I think.

Reply:
Just checking that a tree is deterministically produced by `generate_balanced`. Is that right?

Reply:
Yes, it's fully deterministic.